### PR TITLE
Minor performace improvement when building chunks and rendering blocks

### PR DESCRIPTION
**Pull request by ichttt**
Speed up LightUtil's mapFormats. This method is a hotspot, as it is called for each block that is compiled in a chunk, and the current implementation always creates a Pair, which is then hashed, and then compared via equals, which causes some overhead. But the actual VertexFormats are almost always Block with normals to items (in fact I never saw any other case during my testing), so this can be speed up easily. Some visualvm profiling told me that this is about 3% of the time for compiling chunks was spend on this method, so I think it is worth the little more code.

### CODE REVIEW

**Comment by bs2609** [comment]
I don't agree with the blanket "replace all `values()` calls" approach here - some of these changes are in static initialisers that are run just once.

Personally, either it's worth taking your own copy of the values array, or it's not worth bothering with at all.

**Comment by bs2609** [comment]
As for the second issue, how about something like [this gist](https://gist.github.com/bs2609/923512ae39e92d0884a389178bbde6cf)?

This way the required `equals()` checks can short-circuit on instance identity, which would otherwise never be the case when storing a modified format instance seperately.

**Comment by ichttt** [comment]
I did some mesurements using my [Invocation Profiler](https://gist.github.com/ichttt/88b650ead95b5415c7ee3972cfa123d9), and I got the following results (Everything tested twice):
Current 1.12.x :
MapFormats: Total 137 invocations 1206000 mean 1.1439594E-4 (ms) (+- 10ms total)
This PR (before :
MapFormats: Total 14 invocations 1206000 mean 1.2427482E-5 (ms) (+- 3ms total)
BS:
MapFormats: Total 4 invocations 1200600 mean 3.6181907E-6 (ms) (+- 3ms total)
With Format caching:
MapFormats: Total 1 invocations 1206600 mean 1.5232886E-6 (ms) (+- 2ms total)

As you can see, this method isn't really slow as-is, but because it is called over a million times just after loading a world, this makes a difference. I just went with caching the build vertex format, as this is the solution with the least code and the fastest one, but it doesn't make too much of a difference.

File: src/main/java/net/minecraftforge/client/model/pipeline/BlockInfo.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 public class BlockInfo
 {
-    private static final EnumFacing[] SIDES = EnumFacing.values();
+    private static final EnumFacing[] SIDES = EnumFacing.VALUES;
```
Comment (tterrag1098) [inline src/main/java/net/minecraftforge/client/model/pipeline/BlockInfo.java]: I do not agree with this change - it adds a dependency on mutable state for no gain

File: src/main/java/net/minecraftforge/common/util/InvocationProfiler.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package net.minecraftforge.common.util;
```
Comment (tterrag1098) [inline src/main/java/net/minecraftforge/common/util/InvocationProfiler.java]: I do not think we need this class - it's not a very accurate benchmarker.

**Review by tterrag1098 — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/common/util/InvocationProfiler.java (hunk 2 above)
Comment (ichttt) [reply src/main/java/net/minecraftforge/common/util/InvocationProfiler.java]: Yeah it wasn't meant to be commited. Its my better than nothing profiler

**Comment by bs2609** [comment]
In the case of caching the last-used format mapping, it'd be possible to have this replace the existing map entirely, which would also allow removing the patch to `VertexFormat` to cache the hashcode. Maybe worth considering?

**Review by tterrag1098 — APPROVED** [review]

**Review by mezz — APPROVED** [review]

**Pull request merged**